aws/table_aws_lambda_function_event_source_mapping.go

```diff
@@ -0,0 +1,354 @@
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Topics"),
+			},
+		}),
+	}
+}
+
+//// HYDRATE FUNCTION
+
+func getAwsLambdaFunctionTrigger(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	var uuid string
+
+	if h.Item != nil {
+		uuid = *h.Item.(types.EventSourceMappingConfiguration).UUID
+	} else {
+		uuid = d.EqualsQuals["uuid"].GetStringValue()
+	}
+
+	// Empty input check
+	if strings.TrimSpace(uuid) == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	svc, err := LambdaClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error(EV_SOURCE_MAPPING_TABLE_NAME+".getAwsLambdaFunctionTrigger", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	params := &lambda.GetEventSourceMappingInput{
+		UUID: &uuid,
+	}
+
+	rowData, err := svc.GetEventSourceMapping(ctx, params) // ---> *GetEventSourceMappingOutput
+	if err != nil {
+		plugin.Logger(ctx).Error(EV_SOURCE_MAPPING_TABLE_NAME+".getAwsLambdaFunctionTrigger", "api_error", err)
+		return nil, err
+	}
+
+	return rowData, nil
+
+}
+
+//// LIST FUNCTION
+
+func listAwsLambdaFunctionTriggers(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create service
+	svc, err := LambdaClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error(EV_SOURCE_MAPPING_TABLE_NAME+".listAwsLambdaFunctionTriggers", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	maxItems := int32(10000)
+	input := lambda.ListEventSourceMappingsInput{}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 1 {
+				maxItems = int32(1)
+			} else {
+				maxItems = int32(limit)
+			}
+		}
+	}
+
+	input.MaxItems = aws.Int32(maxItems)
+	paginator := lambda.NewListEventSourceMappingsPaginator(svc, &input, func(o *lambda.ListEventSourceMappingsPaginatorOptions) {
+		o.Limit = maxItems
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error(EV_SOURCE_MAPPING_TABLE_NAME+".listAwsLambdaFunctionTriggers", "api_error", err)
+			return nil, err
+		}
+
+		for _, mapping := range output.EventSourceMappings {
+			d.StreamListItem(ctx, mapping)
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func getFunctionNameFromArn(ctx context.Context, td *transform.TransformData) (interface{}, error) {
+	arn := *td.HydrateItem.(types.EventSourceMappingConfiguration).FunctionArn
+	parts := strings.Split(arn, ":")
+	return parts[len(parts)-1], nil
+}
+
+// Commented these, and the using columns because the implementation is brittle or non-useful in some cases.
```

Comment:
If we don't use these functions, can we remove them and add examples to extract the info, e.g., using the Postgres `split_part()` function?